src/process/volcanic/CCPr_Volcanic_Mod.F90

```diff
@@ -0,0 +1,381 @@
+!> \brief CCPR Volcanice state types
+!!
+!! \defgroup catchem_Volcanic_process
+!!
+!! \author Lacey Holland and Wei Li
+!! \date 10/2024
+!!!>
+MODULE CCPR_Volcanic_mod
+   USE Precision_mod
+   USE Error_Mod
+   USE DiagState_Mod, Only : DiagStateType
+   USE MetState_Mod,  Only : MetStateType
+   USE ChemState_Mod, Only : ChemStateType
+   USE Config_Opt_Mod, Only : ConfigType
+   USE EmisState_Mod, Only: EmisStateType
+
+   IMPLICIT NONE
+
+   PRIVATE
+
+   PUBLIC :: CCPR_Volcanic_Init
+   PUBLIC :: CCPR_Volcanic_Run
+   PUBLIC :: CCPR_Volcanic_Finalize
+   PUBLIC :: VolcanicStateType
+
+
+   !> \brief VolcanicStateType
+   !!
+   !! VolcanicStateType is the process-specific derived type.
+   !!
+   !! \param Activate Activate Process (True/False)
+   !! \param Scheme Scheme Option
+   !! \param nVolcanicSpecies # of Volcanic species
+   !! \param VolcanicSpeciesIndex Index of Volcanic species
+   !! \param VolcanicSpeciesName Name of Volcanic species
+   !! \param SpcIDs CATChem species IDs
+   !! \param CatIndex Index of emission category in EmisState
+   !! \param TotalEmission Total emission of all species at each level [kg/m^2/s]
+   !! \param EmissionPerSpecies Emission per species at each level [kg/m^2/s]
+   !! \param FileDir Input file directory for reading in emissions
+   !!
+   !! \ingroup catchem_Volcanic_process
+   !!!>
+   TYPE :: VolcanicStateType
+
+      ! Generic Variables for Every Process
+      LOGICAL                         :: Activate              ! Activate Process (True/False)
+      INTEGER                         :: SchemeOpt             ! Scheme Option (if there is only one SchemeOpt always = 1)
+      integer                         :: nVolcanicSpecies           !< Number of Volcanic species
+      integer, allocatable            :: VolcanicSpeciesIndex(:)    !< Index of Volcanic species
+      character(len=31), allocatable  :: VolcanicSpeciesName(:)     !< name of Volcanic species
+      integer, allocatable            :: SpcIDs(:)               !< CATChem species IDs
+      integer                         :: CatIndex                !< Index of emission category in EmisState
+
+      ! Process Specific Parameters
+      real(fp), allocatable           :: TotalEmission(:)          !< Total emission of all species at each level [kg/m^2/s]
+      real(fp), allocatable           :: EmissionPerSpecies(:,:)   !< Emission per species at each level          [kg/m^2/s]
+      character(len=1055)             :: FileDir                  !< Input file directory for reading in emissions
+
```

Comment:
Again I would update this to have all capitals so that it looks more consistent. It looks messy to have both INTEGER and integer for example in the same place of the code.